mat/cholesky.go

```diff
@@ -58,6 +71,48 @@ func (c *Cholesky) updateCond(norm float64) {
 	c.cond = 1 / v
 }
 
+// Dims returns the dimensions of the matrix.
+func (chol *Cholesky) Dims() (r, c int) {
+	if !chol.valid() {
+		panic(badCholesky)
+	}
+	r, c = chol.chol.Dims()
+	return r, c
+}
+
+// At returns the element at row i, column j.
+func (c *Cholesky) At(i, j int) float64 {
+	if !c.valid() {
+		panic(badCholesky)
+	}
+	n := c.Symmetric()
+	if uint(i) >= uint(n) {
+		panic(ErrRowAccess)
+	}
+	if uint(j) >= uint(n) {
+		panic(ErrColAccess)
+	}
+
+	var val float64
+	for k := 0; k <= min(i, j); k++ {
+		val += c.chol.at(k, i) * c.chol.at(k, j)
+	}
+	return val
+}
+
+// T implements the Matrix interface. Symmetric matrices, by definition, are
```

Comment:
`T` doesn't strictly speaking implement the `Matrix` interface, it contributes to that.

```
// T returns the the receiver, the transpose of a symmetric matrix.
```